msal/src/Microsoft.Identity.Client/Features/TokenCache/TokenCacheExtensions.cs

```diff
@@ -163,13 +133,28 @@ public static byte[] Serialize(this TokenCache tokenCache)
             // reads the underlying in-memory dictionary and dumps out the content as a JSON
             lock (tokenCache.LockObject)
             {   
-                RequestContext requestContext = new RequestContext(new MsalLogger(Guid.Empty, null));
-                Dictionary<string, IEnumerable<string>> cacheDict = new Dictionary<string, IEnumerable<string>>();
-                cacheDict["access_tokens"] = tokenCache.GetAllAccessTokenCacheItems(requestContext);
-                cacheDict["refresh_tokens"] = tokenCache.GetAllRefreshTokenCacheItems(requestContext);
-                cacheDict["id_tokens"] = tokenCache.GetAllIdTokenCacheItems(requestContext);
-                cacheDict["accounts"] = tokenCache.GetAllAccountCacheItems(requestContext);
-                return JsonHelper.SerializeToJson(cacheDict).ToByteArray();
+                return TokenCacheSerializeHelper.SerializeMsalCache(tokenCache.tokenCacheAccessor);
+            }
+        }
+
+        /// <summary>
+        /// Serializes the entiere token cache
+        /// </summary>
+        /// <param name="tokenCache">Token cache to serialize</param>
+        /// A tuple containing the following information:
+        /// <item><see cref="Tuple{T1,T2}.Item1"/>: Current state of the Unified cache as a blob</item>
+        /// <item><see cref="Tuple{T1,T2}.Item2"/>: Current state of the Adal cache as a blob</item>
+        /// </returns>
+        public static Tuple<byte[], byte[]> SerializeUnifiedAndAdalCache(this TokenCache tokenCache)
```

Comment:
Same thing. since this is part of the public API, could we have a strongly typed structure instead of a tuple?

Reply:
or a "strongly" named tuple? the type is strong imo, it's two byte arrays.

---
In reply to: [211683879](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1217#discussion_r211683879) [](ancestors = 211683879)